plugins/qiskit_executor/routes.py

```diff
@@ -405,20 +391,17 @@ def render(self, db_id, data: Mapping, errors: dict, valid: bool):
             msg = f"Could not load task data with id {db_id} to read parameters!"
             raise KeyError(msg)
 
-        backend_parameter_schema = BackendParameterSchema()
-        backend_parameter_schema.fields["backend"].metadata[
-            "datalist"
-        ] = PluginState.get_value(
-            QiskitExecutor.instance.identifier, f"backend_list_{db_task.id}", default=[]
-        )
+        # provide backend names to template
+        parameter_schema = BackendParameterSchema()
+        parameter_schema.datalists = {"backend-datalist": db_task.data["backend_names"]}
```

Comment:
The key should be exactly the same as for the field (but you can choose either python's snake_case or json's camelCase style, as both are used interchangeable in the schemas). In short, there should be no need for a "-datalist" suffix here.